lib/tasks/evm_dba.rake

```diff
@@ -0,0 +1,260 @@
+require 'awesome_spawn'
+require 'evm_rake_helper'
+
+# TODO: move into DatabaseYml
+# TODO: can we use EvmDatabseOps directly?
+module EvmDba
+  def self.database_configuration_file
+    File.expand_path(File.join(Rails.root, 'config', 'database.yml'))
+  end
+
+  def self.load_config
+    require 'yaml'
+    YAML::load((IO.read(self.database_configuration_file)))
+  end
+
+  def self.local?
+    config = self.load_config[Rails.env]
+    return false unless config['adapter'] == 'postgresql'
+    return %w( 127.0.0.1 localhost ).include?(config['host']) || config['host'].blank?
+  end
+end
+
+namespace :evm do
+  namespace :db do
+    desc 'Start the local ManageIQ EVM Database (VMDB)'
+    task :start do
+      LinuxAdmin::Service.new(PostgresAdmin.service_name).start
+    end
+
+    desc 'Stop the local ManageIQ EVM Database (VMDB)'
+    task :stop do
+      LinuxAdmin::Service.new(PostgresAdmin.service_name).stop
+    end
+
+    # Start the EVM Database silently - not to be a visible rake task
+    task :silent_start do
+      begin
+        LinuxAdmin::Service.new(PostgresAdmin.service_name).start
+      rescue AwesomeSpawn::CommandResultError
+        # ignore issues (ala silent)
+      end
+    end
+
+    # Stop the EVM Database silently - not to be a visible rake task
+    task :silent_stop do
+      begin
+        LinuxAdmin::Service.new(PostgresAdmin.service_name).stop
+      rescue AwesomeSpawn::CommandResultError
+        # ignore issues (ala silent)
+      end
+    end
+
+    desc "Seed the ManageIQ EVM Database (VMDB) with defaults"
+    task :seed do
+      Rake::Task['db:seed'].invoke
+    end
+
+    desc "clean up database"
+    task :gc do
+      require 'trollop'
+      opts = Trollop.options(EvmRakeHelper.extract_command_options) do
+        opt :username,   "Username",         :type => :string
+        opt :password,   "Password",         :type => :string
+        opt :hostname,   "Hostname",         :type => :string
+        opt :dbname,     "Database name",    :type => :string
+        opt :aggressive, "Aggressive gc: vaccume with all options and reindexing"
+        opt :vacuum,     "Vacuum database"
+        opt :reindex,    "Reindex database (or table if --table specified)"
+        opt :analyze,    "Vacuum with analyze"
+        opt :full,       "Vacuum full"
+        opt :verbose,    "Vacuum with verbose information printed"
+
+        opt :table,      "Tablename to reindex (if only perorm on one)", :type => :string
+      end
+
+      opts = opts.delete_if { |_k, v| v.nil? || v == false }
+      EvmDatabaseOps.gc(opts)
```

Comment:
EvmDatabaseOps should probably come over as well.

Reply:
There are a handful of places where we use models in that file. Are we okay with those being in this repo?